lib/src/rules/always_depend_on_packages_you_use.dart

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+import '../ast.dart';
+
+const _desc = r'Depend on packages you use.';
+
+const _details = r'''
+
+**DO** depend on packages you use.
+
+When importing a package, *always* add a dependency on it to your pubspec.
+
+Depending explicitly on packages that you use ensures they will always exist
+and allows you to put a dependency constraint on them to guard you against
+breaking changes.
+
+Whether this should be a regular dependency or dev_dependency depends on if it
+is imported from a public file (one under either `lib` or `bin`), or some other
+file.
+
+**BAD:**
+```dart
+import 'package:a/a.dart';
```

Comment:
Consider removing the import on line 29. It isn't a bad import so it might be confusing to readers.

Reply:
It feels a bit weird having the dependencies totally empty in the pubspec